src/Sylius/Bundle/PayumBundle/Controller/PayumController.php

```diff
@@ -97,7 +99,7 @@ public function prepareCaptureAction(Request $request, $tokenValue)
         $request->getSession()->set('sylius_order_id', $order->getId());
         $options = $configuration->getParameters()->get('redirect');
 
-        $payment = $order->getLastNewPayment();
+        $payment = $order->getLastPayment(PaymentInterface::STATE_NEW);
```

Comment:
As it has been done previously, each operation connected with payment is done on last payment of order. Previously it was always ``NEW`` payment, as it was a default (wrong) state. Now, the default state is ``CART`` and in checkout we should always operate on last ``CART`` payment. However, when checkout is over, we then always operate on ``NEW`` payment - this is required state for payum and also payments with this state are created after previous payment fail/cancel.